### DISCUSSION TITLE
Prevent manual updates for model fields  (createdAt & updatedAt)

### DISCUSSION THREAD

**Discussion by ysfaran** [prisma/prisma#8383] [Q&A] [answered]
So i was searching for a possibility to populate `createdAt` and `updatedAt` fields automatically to a prisma model and came accross a [stackoverflow answer](https://stackoverflow.com/a/65480862/6489012) where this can be acheived with following:

```prisma
model Post {
   id         Int                @id @default(autoincrement())
  // ...
  createdAt   DateTime           @default(now())
  updatedAt   DateTime           @updatedAt
}
```
I was wondering if there is also a way to prevent that prisma client updates these fileds manually: 

```js
prisma.post.update({
      where: {
        id: id,
      },
      data: {
       // if partialPost contains values for createdAt or updateAt they should be ignored
        ...partialPost,
      },
});
```

I could probably do:

```js
prisma.post.update({
      where: {
        id: id,
      },
      data: {
        ...partialPost,
        updatedAt: undefined,
        createdAt: undefined,
      },
});
```
but doing this would be quite cumbersome and error prone.

Is there an easy and more secure way to acheive this?

**Comment by ryands17** [comment, marked as answer]
@ysfaran 👋 

This is not possible as Prisma would always update the default values unless you explicitly specify a value. You can open a request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with your use case so that we can look into this.

**Reply by ysfaran** [reply]
thanks for the advice. 👍 Just submitted a feature request: https://github.com/prisma/prisma/issues/8410

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
